File: app/src/main/java/com/an/trailers/data/remote/model/VideoResponse.java
package com.an.trailers.data.remote.model;

import android.os.Parcel;
import android.os.Parcelable;
import java.util.ArrayList;
import java.util.List;

public class VideoResponse implements Parcelable {

    private long id;
    private List<Video> results;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public List<Video> getResults() {
        return results;
    }

    public void setResults(List<Video> results) {
        this.results = results;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(this.id);
        dest.writeList(this.results);
    }

    public VideoResponse() {
    }

    protected VideoResponse(Parcel in) {
        this.id = in.readLong();
        this.results = new ArrayList<Video>();
        in.readList(this.results, Video.class.getClassLoader());
    }

    public static final Creator<VideoResponse> CREATOR = new Creator<VideoResponse>() {
        @Override
        public VideoResponse createFromParcel(Parcel source) {
            return new VideoResponse(source);
        }

        @Override
        public VideoResponse[] newArray(int size) {
            return new VideoResponse[size];
        }
    };
}

File: app/src/main/java/com/an/trailers/data/remote/model/VideoResponse.kt
package com.an.trailers.data.remote.model

import android.os.Parcel
import android.os.Parcelable

import java.util.ArrayList

class VideoResponse : Parcelable {

    var id: Long = 0
    var results: List<Video>? = null

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(this.id)
        dest.writeList(this.results)
    }

    constructor() {}

    protected constructor(`in`: Parcel) {
        this.id = `in`.readLong()
        this.results = ArrayList()
        `in`.readList(this.results, Video::class.java.classLoader)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<VideoResponse> = object : Parcelable.Creator<VideoResponse> {
            override fun createFromParcel(source: Parcel): VideoResponse {
                return VideoResponse(source)
            }

            override fun newArray(size: Int): Array<VideoResponse?> {
                return arrayOfNulls(size)
            }
        }
    }
}
